mne/io/edf/edf.py

```diff
@@ -242,6 +243,19 @@ def _read_segment_file(self, data, idx, fi, start, stop, cals, mult):
                     assert ch_data.shape == (len(ch_data), buf_len)
                     data[ii, d_sidx:d_eidx] = ch_data.ravel()[r_sidx:r_eidx]
 
+        if subtype == 'bdf':
+            # do not scale stim channel (see gh-5160)
+            if stim_channel is None:
+                stim_idx = [[]]
+            else:
+                stim_idx = np.where(np.arange(self.info['nchan']) ==
+                                    stim_channel)
```

Comment:
why is this different than just doing `stim_idx = stim_channel` ??

```py
(Pdb) stim_channel
0
(Pdb) np.arange(self.info['nchan'])
array([ 0,  1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11, 12, 13, 14, 15, 16,
       17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33,
       34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50,
       51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62, 63, 64])
(Pdb) np.arange(self.info['nchan'])==stim_channel
array([ True, False, False, False, False, False, False, False, False,
       False, False, False, False, False, False, False, False, False,
       False, False, False, False, False, False, False, False, False,
       False, False, False, False, False, False, False, False, False,
       False, False, False, False, False, False, False, False, False,
       False, False, False, False, False, False, False, False, False,
       False, False, False, False, False, False, False, False, False,
       False, False])
(Pdb) np.where(np.arange(self.info['nchan'])==stim_channel)
(array([0]),)
```

Reply:
cc: @cbrnr

Reply:
It is not the same when `stim_channel = None`, which was a possibility before I added the conditional to deal with NumPy comparison to None last week

https://github.com/mne-tools/mne-python/commit/d7c8e865fbfe2dc8854956a446005b05a82300c6

Now I agree it should not be necessary (though if negative indices like `stim_channel=-1` are a possibility here still -- which they probably are not -- then the `np.where` path will actually behave a bit worse)